src/store/models/network.ts

```diff
@@ -922,6 +935,95 @@ const networkModel: NetworkModel = {
 
     actions.setAutoMineMode({ id, mode });
   }),
+  autoBalanceChannels: thunk(async (actions, { id }, { getState, getStoreState }) => {
+    const { networks } = getState();
+    const network = networks.find(n => n.id === id);
+    if (!network) throw new Error(l('networkByIdErr', { id }));
+
+    const getNodeLightningService = (node: LightningNode): LightningService => {
+      switch (node.implementation) {
+        case 'LND':
+          return lndService;
+        case 'c-lightning':
+          return clightningService;
+        case 'eclair':
+          return eclairService;
+        default:
+          throw new Error('unknown implementation');
+      }
+    };
+
+    const balanceChannel = async (
```

Comment:
This function is a bit too long to be declared inside of the action. I would recommend moving this to the [network util](https://github.com/jamaljsr/polar/blob/master/src/utils/network.ts) file instead. This is where I typically keep helper functions for this model.

You can encapsulate most of the logic in this helper function. Giving it the nodes, links, and channels, it can just return an array of objects containing the source, target, and amount to pay for only the rebalances that need to happen. Then here in the store, it just needs to call the RPCs to pay the amounts returned from the helper, if any..

Reply:
awesome idea